libp2p/protocols/pubsub/gossipsub/scoring.nim

```diff
@@ -253,6 +253,12 @@ proc updateScores*(g: GossipSub) = # avoid async
 
   trace "updated scores", peers = g.peers.len
 
+proc scoringHeartbeat*(g: GossipSub) {.async.} =
+  while g.heartbeatRunning:
+    trace "running scoring heartbeat", instance = cast[int](g)
+    g.updateScores()
+    await sleepAsync(g.parameters.decayInterval)
```

Comment:
not the most important thing, but writing sleep loops this way introduces drift - ideally it would do something like `next = next + interval; sleep(now-next)` but guarded against negatives or the scoring will be off (same for heartbeat)

Reply:
Indeed, I would have done the proper loop if there was any await here, but you're right, better safe than sorry
Fixed

Reply:
Actually, the drift was causing issues in CI, which is rather unexpected since this should be a very quick process
Maybe there is some jitter in sleepAsync, or somewhere else

Anyway, I've added a proper heartbeat system, which takes care of the drift properly